azurelinuxagent/common/protocol/wire.py

```diff
@@ -1396,6 +1411,21 @@ def parse(self, xml_text):
             self.ext_handlers.extHandlers.append(ext_handler)
             self.parse_plugin_settings(ext_handler, plugin_settings)
 
+        for ext_handler in self.ext_handlers.extHandlers:
```

Comment:
This is four or five levels of for loops. It is hard for me to track all of things you're trying to accomplish.   Please break this up into well named method or supply comments.

Reply:
Are there test cases for this?